src/vm/vm_core.rs

```diff
@@ -428,8 +445,8 @@ impl VirtualMachine {
 
     pub fn step(&mut self) -> Result<(), VirtualMachineError> {
         if let Some(hint_list) = self.hints.get(&self.run_context.pc) {
-            for hint_code in hint_list.clone().iter() {
-                execute_hint(self, hint_code)?
+            for hint_data in hint_list.clone().iter() {
+                execute_hint(self, &hint_data.hint_code, hint_data.ids.clone())?
```

Comment:
Cloning all the hints data in a loop sounds like a bad idea from a perf point of view.

Reply:
I think it is only the hints list that is not very long. We could merge it and improve it afterwards.